Sources/SwiftParser/Declarations.swift

```diff
@@ -2233,3 +2189,59 @@ extension Parser {
     }
   }
 }
+
+extension Parser {
+  mutating func parseAllEffectsSpecifiers() -> [SpecifierWithToken] {
+    var specifiers = [SpecifierWithToken]()
+    var loopProgress = LoopProgressCondition()
+    while let (specifier, handle) = self.canRecoverTo(anyIn: EffectsSpecifier.self), loopProgress.evaluate(currentToken) {
+      let (unexpected, token) = self.eatKeepUnexpected(handle)
+      specifiers.append(contentsOf: unexpected.map { SpecifierWithToken(specifier: nil, token: $0) })
```

Comment:
I don’t really like how we are considering unexpected tokens as `SpecifierWithToken` here and setting `specifier` to `nil`. 

What do you think about changing `SpecifierWithToken` to 

```swift
struct ParsedEffectsSpecifier {
  let specifier: EffectsSpecifier?
  let unexpected: RawUnexpectedNodesSyntax?
  let token: RawTokenSyntax
 }
```

That way you also don’t need to modify `eat`.

Reply:
Much better name, but I'm not sure I understand the `unexpected` suggestion. The issue here is that they all need to get merged into the single `RawUnexpectedNodeSyntax`.

Reply:
We can add `RawUnexpectedNodeSyntax` as a child of `RawUnexpectedNodeSyntax`. Or, if you don’t like that, maybe it should be 

```swift 
struct ParsedEffectsSpecifier {
  let specifier: EffectsSpecifier?
  let unexpected: [RawTokenSyntax]
  let token: RawTokenSyntax
 }
```

Reply:
Sure, makes sense.